ssh-port-forward/action.yml

```diff
@@ -0,0 +1,81 @@
+name: 'SSH port forwarding action'
+description: 'Forward a connection over SSH in the background'
+
+branding:
+  icon: 'corner-up-right'
+  color: 'red'
+
+inputs:
+  ssh-key:
+    description: 'SSH private key'
+    required: true
+  ssh-host:
+    description: 'SSH host'
+    required: true
+  ssh-port:
+    description: 'SSH port'
+    required: false
+    default: '22'
+  ssh-user:
+    description: 'SSH user'
+    required: true
+  remote-host:
+    description: 'Remote host'
+    required: true
+  remote-port:
+    description: 'Remote port'
+    required: true
+  local-port:
+    description: 'Local port'
+    required: true
+  run-in-container:
+    description: 'Set to true if the action will run inside a container using GitHub container parameter'
+    default: 'false'
+    required: true    
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Start SSH port forwarding
+      shell: bash
+      run: |
+        # Add SSH key
+        mkdir -p ~/.ssh
+        printf "%s" "${{ inputs.ssh-key }}" | tee user_key
+        chmod 400 user_key
+        
+        # Start SSH forwarder
+        ssh -i user_key -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -N -L ${{ inputs.local-port }}:${{ inputs.remote-host }}:${{ inputs.remote-port }} -p ${{ inputs.ssh-port }} ${{ inputs.ssh-user }}@${{ inputs.ssh-host }} &
+
+        # wait until connections are accepted
+        sleep 3
+        isready=0
+
+        waitScriptPath="${{ github.action_path }}"
+        if [[ "${{ inputs.run-in-container }}" -eq "true" ]]; then
+          waitScriptPath="/__w/_actions/jahia/jahia-modules-action/ssh-port-forward/ssh-port-forward"
```

Comment:
This will need to be updated to `/__w/_actions/jahia/jahia-modules-action/v2/ssh-port-forward` before merge, for now this is pointing to the `ssh-port-forward` branch.